packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -125,6 +126,7 @@ async function ssrTransformScript(
     // import * as ok from 'foo' --> ok -> __import_foo__
     if (node.type === 'ImportDeclaration') {
       const importId = defineImport(node.source.value as string)
+      importIdMap.set(node.source.value as string, importId)
```

Comment:
Nice optimization! Should we move this `importIdMap` cache inside `defineImport` instead? So that `defineImport` handles the caching by itself and the consumer doesn't have to check manually.

Reply:
Sure, we can do it. But https://github.com/vitejs/vite/pull/14441 may break this change

Reply:
Hmm yeah I guess it conflicts a bit, but I assume it should be safe to share the cache still for different `index`? Since the first call would always execute first.

Reply:
Yes, just imports need hoisting, but exports do not need to. If this PR merged first. We may need to process all imports first In #14441.

Reply:
I ran some tests and now I'm confused if this is safe. https://stackblitz.com/edit/node-muh71q?file=index.js

Looks like #14441 affects the execution order of imports too, but it was also wrong before anyways. SSR transform has always been head-banging 😵

Reply:
Yes, Let's fix it.